Lib/test/test_subclassinit.py

```diff
@@ -114,11 +114,10 @@ class A:
 
     def test_set_name_metaclass(self):
         class Meta(type):
-            def __new__(cls, name, bases, ns):
-                ret = super().__new__(cls, name, bases, ns)
-                self.assertEqual(ret.d.name, "d")
-                self.assertIs(ret.d.owner, ret)
-                return 0
+            def __init__(this, name, bases, ns):
+                super().__init__(name, bases, ns)
+                self.assertEqual(this.d.name, "d")
+                self.assertIs(this.d.owner, this)
```

Comment:
move tests into `__init__` after `super().__init__()` has run `__init_subclass__` and `__set_name__`